lib/imageflow/graph_runner.ex

```diff
@@ -1,14 +1,36 @@
 defmodule Imageflow.GraphRunner do
-  alias Imageflow.{Graph, Native}
+  alias Imageflow.{Graph, Native, Result}
 
   def run(%Graph{} = graph) do
     with {:ok, job} <- Native.create(),
          :ok <- add_inputs(job, graph.inputs),
          :ok <- add_outputs(job, graph.outputs),
          :ok <- send_task(job, graph),
-         :ok <- save_outputs(job, graph.outputs),
-         :ok <- Native.destroy(job) do
```

Comment:
It seems this step got left out entirely.
I'm not sure where to fit it now, since I'm assuming we need to do this only after `get_results`, but it doesn't seem correct to never de-allocate the job